common4j/src/main/com/microsoft/identity/common/java/nativeauth/controllers/results/INativeAuthCommandResult.kt

```diff
@@ -62,7 +70,19 @@ interface INativeAuthCommandResult {
         SignUpSubmitCodeCommandResult, SignUpResendCodeCommandResult,
         SignUpSubmitPasswordCommandResult,
         ResetPasswordStartCommandResult, ResetPasswordSubmitCodeCommandResult,
-        ResetPasswordResendCodeCommandResult, ResetPasswordSubmitNewPasswordCommandResult
+        ResetPasswordResendCodeCommandResult, ResetPasswordSubmitNewPasswordCommandResult {
+        override fun toSafeString(mayContainPii: Boolean): String {
+            return if (mayContainPii) {
+                "UnknownError(correlationId=$correlationId, error=$error, errorDescription=$errorDescription), details=$details, errorCodes=$errorCodes"
+            } else {
+                "UnknownError(correlationId=$correlationId)"
+            }
+        }
+
+        override fun containsPii(): Boolean = true
```

Comment:
Is there a use case for this flag to be set after the object has been constructed? It would be better if this value is set in the contructor and read only after that.

Reply:
I'm not sure I'm following. This method isn't a setter, but a getter. It isn't set, really. It's a "constant" that will always return true or false.

Depending on which interface/abstract class hierarchy we choose, we can make it a field in the constructor as well.

Reply:
Please resolve if addressed @SaurabhMSFT